### DISCUSSION TITLE
Inline json creation without struct

### DISCUSSION THREAD

**Discussion by gelldur** [beached/daw_json_link#390] [Q&A] [answered]
Hello, is there way to create inline json?
Example:
```cpp
	daw::json::json_value v;
	v["key"] = "value";
```

I couldn't find any example in documentation and of course above code do not compile. I search in code any signs of such approach bot couldn't find.

**Comment by beached** [comment, marked as answer]
Right, not a design goal of the library to have a vocab JSON type. Usually one has local data structures they want to (de)serialize.  In a pinch something like 

```cpp  
auto json = daw::json::to_json( std::map<std::string, std::string> {
  { "key", "value" }
} );
```
https://jsonlink.godbolt.org/z/rWdxG6jKh
would give the json of something like that without mapping a class

But that way of serializing will be painful.  Generally one wants something like 

```cpp
struct Foo {
  std::string name;
  std::string name2;
}
``` 
Locally so that working with it is like any other C++ data type and generally optimal.  Then map it to a JSON object 

```cpp
namespace daw::json {
  template<>
  struct json_data_contract<Foo> {
    using member_list = json_member_list<json_string<"name">, json_string<"name2">>;

   static auto to_json_data( Foo const & f ) {
     return std::forward_as_tuple( f.name, f.name2 );
  };
}
```

For small toy examples it is a bit more, but once one gets to larger systems, it scales nicely and requires little to no code for the mappings.  Another option for simple things, and in the future with reflection will be easier, is to use a library like Boost.Describe on your data structures and JSON Link will work with it, just by adding an include for <daw/json/daw_json_link_describe.h>

**Reply by gelldur** [reply]
Thanks, I agree that this library target different approach, but it is nice to have 1 json lib in project :) Thanks.
